### DISCUSSION TITLE
metric with empty label value will add a new metric record

### DISCUSSION THREAD

**Discussion by lowang-bh** [prometheus/prometheus#14117] [Q&A]
### What did you do?

I have a metric of cluster node's resource with several label, eg: label1, label2,  label1's value won't be empty, but value of label2 can be empty.
At first, the node has label2 with value2, so does the metric.
After I remove the label2 from node, then there is another metric wihout label2, instead of empty string value of label2 in the origin metric(I set the value of empty string to the metric if value is empty).

![image](https://github.com/prometheus/prometheus/assets/21003791/26ffae4e-96ab-4606-83bf-d2be18ce6c46)

### What did you expect to see?

before:
metric{label1="value1", label2="value2"} 1
after value2 change to ""
metric{label1="value1", label2=""}  1

### What did you see instead? Under which circumstances?

metric{label1="value1", label2="value2"} 1
metric{label1="value1"}  1

### System information

Linux 3.10.0-1062.9.1.el7.x86_64 x86_64

### Prometheus version

```text
Version	2.30.2
Revision	b30db03f35651888e34ac101a06e25d27d15b476
Branch	HEAD
BuildUser	root@80d1e84a93dd
BuildDate	20211001-12:26:58
GoVersion	go1.17.1
```

### Prometheus configuration file

```yaml
- args:
    - --storage.tsdb.retention=21d
    - --storage.tsdb.path=/prometheus
    - --config.file=/etc/prometheus/prometheus.yml
    - --web.enable-lifecycle
    - --web.enable-admin-api
```
```


### Alertmanager version

```text
0.14.0
```

**Comment by machine424** [comment]
For Prometheus `Labels with an empty label value are considered equivalent to labels that do not exist.`
https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels

**Comment by lowang-bh** [comment]
> https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels

- The change of any label's value, including adding or removing labels, will create a new time series.
- Labels with an empty label value are considered equivalent to labels that do not exist.

Thanks. I see it . However, how can I make the old metric invalid. Now the sum value of the metrics with label1 will be double.

**Comment by machine424** [comment]
What do you mean by `invalid`? Could you elaborate?

**Reply by lowang-bh** [reply]
I means to let the old metric does not appear any more, or let its value to be zero after label2's value is changed.

example: 

before remove label2's value

- metric{label1="value1", label2="value2"} 1

after change label2's value to empty,

- metric{label1="value1", label2="value2"} 0
- metric{label1="value1"} 1

so that sum(metric{label1="value1"}) is still 1.

**Comment by machine424** [comment]
I'll turn this into a discussion though.
